packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1749,8 +1760,9 @@ Run 'flutter -h' (or 'flutter <command> -h') for available flutter commands and
 
     if (commandPath != null) {
       Usage.command(commandPath, parameters: CustomDimensions(
-        commandHasTerminal: globals.stdio.hasTerminal,
+        commandHasTerminal: hasTerminal,
       ).merge(await usageValues));
+      analytics.send(await unifiedAnalyticsUsageValues(commandPath));
```

Comment:
What about doing:

```dart
    if (commandPath != null) {
      final List<Object> pairOfUsageValues = await Future.wait<Object>(<Future<Object>>[
        usageValues,
        unifiedAnalyticsUsageValues(commandPath),
      ]);
      Usage.command(commandPath, parameters: CustomDimensions(
        commandHasTerminal: hasTerminal,
      ).merge(pairOfUsageValues[0] as CustomDimensions));
      analytics.send(pairOfUsageValues[1] as Event);
    }
```

To avoid generating one set of values blocking the other.